polus-feature-extraction-3d-plugin/src/main.py

```diff
@@ -0,0 +1,1125 @@
+from skimage import measure
+from skimage.measure import shannon_entropy
+from skimage.segmentation import clear_border
+from scipy.stats import skew
+from scipy.stats import kurtosis as kurto
+from scipy.stats import mode as modevalue
+from scipy.spatial import ConvexHull
+from bfio import BioReader,BioWriter
+from multiprocessing import cpu_count
+from scipy import special,ndimage,signal
+import numpy as np
+import pandas as pd
+import skimage.morphology
+import argparse
+import logging
+import os
+import math
+import sys
+import filepattern
+import itertools
+import tempfile
+
+# Initialize the logger
+logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+					datefmt='%d-%b-%y %H:%M:%S')
+logger = logging.getLogger("main")
+logger.setLevel(logging.INFO)
+
+def read(img_file):
+    """Read the .ome.tif image using BioReader.
+    
+    Args:
+        img_directory (str): Path to the directory containing the input images.
+        
+    Returns:
+        Array of the image and the embedded unit in the metadata if present else it will be none.
+        
+    """
+    tile_grid_size = math.ceil(math.sqrt(cpu_count()))
+    chunk_size = tile_grid_size * 1024
+    br = BioReader(img_file)
+    print(br.Y)
+    print(yyy)
+    bfshape = br.shape
+    img_unit = br.ps_y[1]
+    img_data = []
+    # Loop through timepoints
+    for t in range(br.T):
+        # Loop through channels
+        for c in range(br.C):
+            # Loop through z-slices
+            for z in range(br.Z):
+                # Loop across the length of the image
+                for y in range(0,br.Y,chunk_size):
+                    y_max = min([br.Y,y+chunk_size])
+                    # Loop across the depth of the image
+                    for x in range(0,br.X,chunk_size):
+                        x_max = min([br.X,x+chunk_size])
+                        data_tile = br[y:y_max,x:x_max,z:z+1,c,t]
+                        data_tile = data_tile.flatten()
+                        img_data = np.append(img_data,data_tile)
+    img_data = np.reshape(img_data,[bfshape[0],bfshape[1],bfshape[2]])
+    label_image = img_data.astype(int)
```

Comment:
Why not just load the data directly with bfio rather than looping through things?